server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3835,14 +3836,51 @@ private record CreateSnapshotTask(
 
         @Override
         public void onFailure(Exception e) {
-            logger.warn(
-                () -> format("[%s][%s] failed to create snapshot", snapshot.getRepository(), snapshot.getSnapshotId().getName()),
-                e
-            );
+            final var logLevel = snapshotFailureLogLevel(e);
+            if (logLevel == Level.INFO && logger.isDebugEnabled() == false) {
+                // suppress stack trace at INFO unless extra verbosity is configured
+                logger.info(
+                    format(
+                        "[%s][%s] failed to create snapshot: %s",
+                        snapshot.getRepository(),
+                        snapshot.getSnapshotId().getName(),
+                        e.getMessage()
+                    )
+                );
+            } else {
+                logger.log(
+                    logLevel,
+                    () -> format("[%s][%s] failed to create snapshot", snapshot.getRepository(), snapshot.getSnapshotId().getName()),
+                    e
+                );
+            }
```

Comment:
Good one!

I think it is becoming a pattern to conditionally log stack trace depending on some criteria.
I wonder if it is worth having a common utility to avoid repeating the message formatting and adding wide if-else in different places?

Something like: `log(logger, level, messageFormatter, includeException, exception)`.

Reply:
Or may be 
```
                logger.log(
                    logLevel,
                    () -> format("[%s][%s] failed to create snapshot", snapshot.getRepository(), snapshot.getSnapshotId().getName()),
                     logger.isDebugEnabled() || logLevel != Level.INFO ? e : null
                );
```
works the same way?

Reply:
Quite possibly we should generalize this, although that's not quite the pattern. We still report the exception message at INFO level, just folded into the log message rather than on the next line. Possibly we could always do that by just appending it to the message separated by `": "` or possibly we need two formatters.

Feel free to pursue that change if you'd like.